src/main/java/org/folio/circulation/resources/OverrideCheckOutByBarcodeResource.java

```diff
@@ -0,0 +1,82 @@
+package org.folio.circulation.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.domain.representations.CheckOutByBarcodeRequest.COMMENT;
+import static org.folio.circulation.domain.representations.CheckOutByBarcodeRequest.DUE_DATE;
+import static org.folio.circulation.domain.representations.CheckOutByBarcodeRequest.ITEM_BARCODE;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.policy.library.ClosedLibraryStrategyService;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.server.ValidationError;
+import org.joda.time.DateTime;
+
+import io.vertx.core.http.HttpClient;
+
+public class OverrideCheckOutByBarcodeResource extends CheckOutByBarcodeResource {
+
+  public OverrideCheckOutByBarcodeResource(HttpClient client) {
+    super(client, "/circulation/override-check-out-by-barcode");
+  }
+
+  @Override
+  CompletableFuture<Result<LoanAndRelatedRecords>> applyLoanPolicy(LoanAndRelatedRecords relatedRecords,
```

Comment:
This method does not seem to apply a loan policy.

Was this inheritance added to reduce duplication between a check out and an overridden check out?

(I introduced the inheritance of resources for performing an action using either an ID or a barcode, in retrospect, this was a poor abstraction and has encouraged it's use in other places)

Reply:
Refactored _CheckOutByBarcodeResource_ to replace inheritance with aggregation.

Reply:
Thanks